From 6ec2b4272f802668b11ca450dee2df4aacfa2819 Mon Sep 17 00:00:00 2001
From: "locietta@gmail.com" <locietta@gmail.com>
Date: Fri, 27 Feb 2026 17:30:01 +0530
Subject: [PATCH] Revert "fuse: fix livelock in synchronous file put from
 fuseblk workers"

Revert the previous change that always used asynchronous
fuse_file_put, because it breaks the wslg's virtio.

Signed-off-by: Locietta <locietta@gmail.com>
---
 fs/fuse/file.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index 3b2a171e652f0c9dd1c9e37253d3d3e88caab148..4f7b35cf52d4acb71265bdc114a313f68c537e95 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -368,14 +368,8 @@ void fuse_file_release(struct inode *inode, struct fuse_file *ff,
 	 * Make the release synchronous if this is a fuseblk mount,
 	 * synchronous RELEASE is allowed (and desirable) in this case
 	 * because the server can be trusted not to screw up.
-	 *
-	 * Always use the asynchronous file put because the current thread
-	 * might be the fuse server.  This can happen if a process starts some
-	 * aio and closes the fd before the aio completes.  Since aio takes its
-	 * own ref to the file, the IO completion has to drop the ref, which is
-	 * how the fuse server can end up closing its clients' files.
 	 */
-	fuse_file_put(ff, false);
+	fuse_file_put(ff, ff->fm->fc->destroy);
 }
 
 void fuse_release_common(struct file *file, bool isdir)
